File: PolicyPlus/ConfigurationStorage.vb
Imports Microsoft.Win32
Public Class ConfigurationStorage
    Dim ConfigKey As RegistryKey
    Public Sub New(RegistryBase As RegistryHive, Subkey As String)
        Try
            ConfigKey = RegistryKey.OpenBaseKey(RegistryBase, RegistryView.Default).CreateSubKey(Subkey)
        Catch ex As Exception
            ' The key couldn't be created
        End Try
    End Sub
    Public Function GetValue(ValueName As String, DefaultValue As Object) As Object
        If ConfigKey IsNot Nothing Then Return ConfigKey.GetValue(ValueName, DefaultValue) Else Return DefaultValue
    End Function
    Public Sub SetValue(ValueName As String, Data As Object)
        If ConfigKey IsNot Nothing Then ConfigKey.SetValue(ValueName, Data)
    End Sub
    Public Function HasValue(ValueName As String) As Boolean
        Return ConfigKey IsNot Nothing AndAlso ConfigKey.GetValue(ValueName) IsNot Nothing
    End Function
End Class

File: PolicyPlus/ConfigurationStorage.cs
using System;
using Microsoft.Win32;

namespace PolicyPlus
{
    public class ConfigurationStorage
    {
        private RegistryKey ConfigKey;
        public ConfigurationStorage(RegistryHive RegistryBase, string Subkey)
        {
            try
            {
                ConfigKey = RegistryKey.OpenBaseKey(RegistryBase, RegistryView.Default).CreateSubKey(Subkey);
            }
            catch (Exception ex)
            {
                // The key couldn't be created
            }
        }
        public object GetValue(string ValueName, object DefaultValue)
        {
            if (ConfigKey is not null)
                return ConfigKey.GetValue(ValueName, DefaultValue);
            else
                return DefaultValue;
        }
        public void SetValue(string ValueName, object Data)
        {
            if (ConfigKey is not null)
                ConfigKey.SetValue(ValueName, Data);
        }
        public bool HasValue(string ValueName)
        {
            return ConfigKey is not null && ConfigKey.GetValue(ValueName) is not null;
        }
    }
}